posthog/queries/base.py

```diff
@@ -179,7 +179,14 @@ def empty_or_null_with_value_q(
 ) -> Q:
 
     if operator == "exact" or operator is None:
-        target_filter = lookup_q(f"{column}__{key}", Property._parse_value(value))
+        value_as_given = Property._parse_value(value)
+        value_as_coerced_to_number = Property._parse_value(value, convert_to_number=True)
+        if value_as_given == value_as_coerced_to_number:
```

Comment:
What is this for again? Doesn't the else statement encompass this case too? 😮

Reply:
Can add a comment, tell me more about what's confusing here?

It's just an optimisation to not have an OR statement between when both values are the same

Reply:
oh but if the value is a number and there exists a record for it, won't it return before evaluating the second half of the `|` statement? Or it always evaluates both

Reply:
SQL evaluation is not shortcircuited, the where clauses are unpredictable (i.e. the first where clause might not be the one that's evaluated first)

Reply:
this Q object generates a SQL in the end, and that SQL is passed to the db, which then returns on appropriate match; the more concise we can make this SQL, the better.

Reply:
TIL! thanks 🤗